### DISCUSSION TITLE
question in debugging

### DISCUSSION THREAD

**Discussion by wendy444444** [yalmip/YALMIP#1125] [General discussion]
When I used Matlab + Yalmip+ Gurobi to solve the problem, my program suggested that the model was infeasible or unbounded, so I separated the constraints according to https://yalmip.github.io/debugginginfeasible. 
But why did this situation occur in the picture below? Before, the results appeared quickly, but this time, it took a lot of time to like to solve the optimization results in the solver. I don't understand is this right? What am I supposed to do if not?
![mmexport1653371245764](https://user-images.githubusercontent.com/103562806/169957968-db8a0c29-5cd4-43f6-ae80-00f9baa1559b.png)

**Comment by johanlofberg** [comment]
The problem appears extremely badly scaled. It appears gurobi finds a solution to scaled problem, but then when going back to original model, it becomes infeasible. It just looks like a horribly bad model, i.e. if you look at C you will likely see some constraints have obscenely large coefficients

**Reply by wendy444444** [reply]
Sir, thank you for your answer. Excuse me, is that because my explicit variable bounds  is set too big or other constraints?

**Reply by wendy444444** [reply]
> The problem appears extremely badly scaled. It appears gurobi finds a solution to scaled problem, but then when going back to original model, it becomes infeasible. It just looks like a horribly bad model, i.e. if you look at C you will likely see some constraints have obscenely large coefficients

I think there may be something wrong with the setting of variables. What I think may go wrong is:
```
numbus=12;
tt=10;
Eplcon=600;
V_12=sdpvar(numbus-1,tt,'full');
V22=[UB*ones(1,tt);V_12];
 C = [C,((0<=V22)&(V22<=Eplcon*Eplcon)):'V22'];
 C = [C,((0<=V_12)&(V_12<=Eplcon*Eplcon)):'V_12'];
```
That's because Matlab gave a warning at V22: variable might be set by a nonscalar operator.
However, the constraints I checked didn't include these two constraints, but the variable V22 was in the latter constraints.

**Reply by wendy444444** [reply]
Sir, I added Explicit variable bounds to all constraints. Is this right? I read warning.https://yalmip.github.io/faq/warningbigm/ because I had received a warning. I guess there's a problem here.Could you please help me?

**Reply by johanlofberg** [reply]
https://yalmip.github.io/inside/debuggingnumerics/. The complaints on numerics indicate far worse numbers (10^11) than what would be added to a model with missing explicit bounds (10^4). And obviously you got some weird shit going on with that warning, and should thus look at exactly which operation causes that warning and what the involved expressions look like. Just don't read the warning without digging into it

btw, your code looks messy as you use both , and & to separate constraints., and brackets on some and parenthesis on some. . A more cleaner code  `C = [C, [0<=V22, V22<=Eplcon*Eplcon]:'V22'];`

**Reply by wendy444444** [reply]
Thank you, sir. It's worked.
